integration/error-sanitization-test.ts

```diff
@@ -0,0 +1,358 @@
+      );
+    });
+
+    test("sanitizes loader errors in resource requests", async () => {
+      let response = await fixture.requestData(
+        "/resource?loader",
+        "routes/resource"
+      );
+      let text = await response.text();
+      expect(text).toBe('{"message":"Unexpected Server Error"}');
+    });
+
+    test("sanitizes mismatched route errors in data requests", async () => {
+      let response = await fixture.requestData("/", "not-a-route");
+      let text = await response.text();
+      expect(text).toBe('{"message":"Unexpected Server Error"}');
+    });
+  });
+
+  test.describe("serverMode=development", () => {
+    test.beforeAll(async () => {
+      fixture = await createFixture(
+        {
+          files: routeFiles,
+        },
+        ServerMode.Development
+      );
+    });
+
+    test("renders document without errors", async () => {
+      let response = await fixture.requestDocument("/");
+      let html = await response.text();
+      expect(html).toMatch("Index Route");
+      expect(html).toMatch("LOADER");
+      expect(html).not.toMatch("MESSAGE:");
+      expect(html).not.toMatch(/stack/i);
+    });
+
+    test("does not sanitize loader errors in document requests", async () => {
+      let response = await fixture.requestDocument("/?loader");
+      let html = await response.text();
+      expect(html).toMatch("Index Error");
+      expect(html).not.toMatch("LOADER");
+      expect(html).toMatch("<p>MESSAGE:Loader Error");
+      expect(html).toMatch("<p>STACK:Error: Loader Error");
+      expect(html).toMatch(
+        'errors":{"routes/index":{"message":"Loader Error","stack":"Error: Loader Error\\n'
+      );
+    });
+
+    test("does not sanitize render errors in document requests", async () => {
+      let response = await fixture.requestDocument("/?render");
+      let html = await response.text();
+      expect(html).toMatch("Index Error");
+      expect(html).toMatch("<p>MESSAGE:Render Error");
+      expect(html).toMatch("<p>STACK:Error: Render Error");
+      expect(html).toMatch(
+        'errors":{"routes/index":{"message":"Render Error","stack":"Error: Render Error\\n'
+      );
+    });
+
+    test("renders deferred document without errors", async () => {
+      let response = await fixture.requestDocument("/defer");
+      let html = await response.text();
+      expect(html).toMatch("Defer Route");
+      expect(html).toMatch("RESOLVED");
+      expect(html).not.toMatch("MESSAGE:");
+      expect(html).not.toMatch(/stack/i);
+    });
+
+    test("does not sanitize defer errors in document requests", async () => {
+      let response = await fixture.requestDocument("/defer?loader");
+      let html = await response.text();
+      expect(html).toMatch("Defer Error");
+      expect(html).not.toMatch("RESOLVED");
+      expect(html).toMatch('{"message":"REJECTED"}');
+      // TODO: I think we should be getting the stack here too?
+      expect(html).toMatch(/stack/i);
```

Comment:
@jacob-ebey Should we get the stack here?  I think it might have something to do with not having `NODE_ENV` set right in the E2E setup?